File: src/1876_Substrings_of_Size_Three_with_Distinct_Characters/solution.py
## https://leetcode.com/problems/substrings-of-size-three-with-distinct-characters/


def countGoodSubstrings(s: str) -> int:
    window_size = 3
    l = 0
    r = 0
    n = len(s)
    count = 0
    while r < n:
        if r - l + 1 < window_size:
            r += 1
        elif r - l + 1 == window_size:
            if s[l] != s[l + 1] and s[l] != s[l + 2] and s[l + 1] != s[l + 2]:
                count += 1
            l += 1
            r += 1
    return count

File: src/1876_Substrings_of_Size_Three_with_Distinct_Characters/solution.ts
// https://leetcode.com/problems/substrings-of-size-three-with-distinct-characters/

function countGoodSubstrings(s: string): number {
  const window_size = 3;
  let l = 0;
  let r = 0;
  const n = s.length;
  let count = 0;
  while (r < n) {
    if (r - l + 1 < window_size) {
      r += 1;
    } else if (r - l + 1 == window_size) {
      if (s[l] != s[l + 1] && s[l] != s[l + 2] && s[l + 1] != s[l + 2]) {
        count += 1;
      }
      l += 1;
      r += 1;
    }
  }
  return count;
}
